index.js

```diff
@@ -371,17 +371,21 @@ class HtmlWebpackPlugin {
     if (templateParameters === false) {
       return Promise.resolve({});
     }
+    const preparedAssetTags = {
+      headTags: this.prepareAssetTagGroupForRendering(assetTags.headTags),
+      bodyTags: this.prepareAssetTagGroupForRendering(assetTags.bodyTags)
+    };
     if (typeof templateParameters === 'function') {
-      const preparedAssetTags = {
-        headTags: this.prepareAssetTagGroupForRendering(assetTags.headTags),
-        bodyTags: this.prepareAssetTagGroupForRendering(assetTags.bodyTags)
-      };
       return Promise
         .resolve()
         .then(() => templateParameters(compilation, assets, preparedAssetTags, this.options));
     }
-    if (typeof templateParameters === 'object') {
-      return Promise.resolve(templateParameters);
+    if (templateParameters && typeof templateParameters === 'object') {
```

Comment:
Because `templateParameters` may be `null`, and `typeof null === 'object'` will be `true`. So I think use `templateParameters && typeof templateParameters === 'object'` will be safer.

Reply:
`null` would be a strange configuration - but are there any possible problems if it would be `null`?

It seems to be mergeable with your code:

```
console.log({...{a: 1}, ...null}); // -> {a: 1}
```

Reply:
Oh~~, it's amazing. I used to think this will cause an error.

I just don't know that spread syntax will handle null and undefined properly, and thank you.

So, `if (templateParameters && typeof templateParameters === 'object')` can be simplified to `if (typeof templateParameters === 'object')`.